In [3]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression, Lasso, Ridge, LassoCV, BayesianRidge
import statsmodels.formula.api as sm
import matplotlib.pylab as plt
import math
from scipy.stats import norm

# Newsletter Signup Rates (A/B Test)
You work for a website that’s testing two different landing pages (A and B) to see which one drives more newsletter signups.
Page A: Out of 300 visitors, 42 signed up for the newsletter.
Page B: Out of 320 visitors, 56 signed up for the newsletter.

## Question:
Is there a statistically significant difference in the signup rates between the two pages at a 0.05 significance level?

#### Hypotheses
* **Null:** pageA = pageB
* **Alternative:** = pageA - pageB > 0
* **Z-Critical** = 1.65
* **alpha** = .05

In [8]:
phatA = 42/300
phatB = 56/320

In [9]:
pooled_p = (42 + 56) / (300 + 320)
pooled_p

0.15806451612903225

In [10]:
SE = math.sqrt((pooled_p * (1-pooled_p)) * ((1/300) + (1/320)))
SE

0.02931681780909403

In [11]:
z = abs((phatA - phatB))/ SE
z
p_value = 2* (1 - .8830) #wanted to the right of the p-value
print(f"Z-Score: {z}\nP-Value: {p_value}")

Z-Score: 1.1938539928826462
P-Value: 0.23399999999999999


## Fail to reject the null hypothesis
The P-Value is .234 and the z-score is 1.19. And there is no significant difference in the signup rates between the two.  

# Vaccine Efficacy in Two Cities
A new flu vaccine was rolled out in two cities. After flu season:
* City 1: Out of 800 vaccinated people, 40 still caught the flu.
* City 2: Out of 900 vaccinated people, 27 caught the flu.

## Question:
Is there evidence that the vaccine worked better in one city compared to the other? Use a 2-proportion z-test at a 0.05 significance level.

* **Null Hypothesis:** city1 - city2 = 0
* **Alternate Hypothesis:** city1 - city2 != 0
* **Z-Critical:** 1.96
* **Alpha:** .05

In [1]:
city1 = 40 / 800
city2 =  27/ 900
print(f"For City 1: {city1} percent still caught the flu\nFor City 2: {city2} percent still caught the flu")

For City 1: 0.05 percent still caught the flu
For City 2: 0.03 percent still caught the flu


In [2]:
check = city1 - city2
check

0.020000000000000004

### Which city did better?
In the population samples for CITY 2 and CITY 1, in CITY 2 3 percent of those vaccinated caught the flu while for CITY 1 5 percent of those vaccinated caught the flu. There is a 2 percent difference between those that caught the flu after vaccination in both cities.



In [3]:
pooled_p = (40 + 27) / (800 + 900)
pooled_p

0.039411764705882354

In [14]:
SE = math.sqrt((pooled_p *(1-pooled_p)) * ((1/800) + (1/900)))
SE

0.009454526529415678

In [19]:
z = abs(city1 - city2)/SE
z
p_value = 2*(1 - .9830)
p_value
print(f"Z-Score:{z}\nP-value:{p_value}")

Z-Score:2.1153888497509006
P-value:0.03400000000000003


### Reject the null hypothesis
There is evidence that the vaccine worked in City 1 over City 2.
* **Z-statistic** is 2.12 standard deviations above what is expected if the null hypothesis were true.
* **Standard Errors** With an SE of .01, I can be confident about the difference.
* **P-value** is 0.03, which is less than alpha. We have 3 percent chance of seeing a z-statistic greater than the null hypothesis. 